src/panels/config/helpers/forms/ha-input_datetime-form.ts

```diff
@@ -35,6 +35,8 @@ class HaInputDateTimeForm extends LitElement {
           : item.has_time
           ? "time"
           : "date";
+      this._item.has_date =
+        !item.has_date && !item.has_time ? true : item.has_date;
```

Comment:
Should this perhaps be above the `this._mode` determination right before?

Reply:
Can't this check just be `mode != "time` ?

Reply:
mode is not on the item